apps/api-extractor/src/collector/Collector.ts

```diff
@@ -485,10 +487,14 @@ export class Collector {
         entity.singleExportName !== undefined &&
         entity.singleExportName !== ts.InternalSymbolName.Default
       ) {
-        idealNameForEmit = entity.singleExportName;
-      } else {
+        // TODO: remove replace after "exportPath" support
```

Comment:
@javier-garcia-meteologica Could you provide some more information about this TODO?  What do you mean by "exportPath" support?

Reply:
I'm referring to [adventure-yunfei#4](https://github.com/adventure-yunfei/web-build-tools/pull/4) and the suggestion to use `"exportPath"` on `"exportName"` for `AstImport`.

The last update to my PR introduced some changes: `import() types` are given their own `AstImportKind` (`AstImportKind.ImportType`) and the complete export path is put into `exportName` as a string.

For instance, `import('foo').X.Y.Z` produces:

```
AstImport {
  importKind: AstImportKind.ImportType,
  modulePath: 'foo',
  exportName: 'X.Y.Z'
}
```

This was coded with `"exportPath"` support in mind. After `"exportPath"` support is merged, it won't be necessary to parse `"exportName"` using `exportName.split('.')` or transform it using `exportName.replace(...)`, because that information will already be available in `AstImport`.

```
AstImport {
  importKind: AstImportKind.ImportType,
  modulePath: 'foo',
  exportPath: ['X', 'Y'],
  exportName: 'Z'
}
```

After `"exportPath"` support `"singleExportName"` just contains `"Z"`, instead of `"X.Y.Z"`.

```
        // idealNameForEmit = entity.singleExportName.replace(/\./g, '_');
        idealNameForEmit = entity.singleExportName;
```

Reply:
I added [a test case](https://github.com/microsoft/rushstack/pull/1916/files#diff-d3890b10e71ee967c150e8edde583bfe) to your branch that takes this input:

**api-extractor-scenarios/src/dynamicImportType2/index.ts**
```ts
/** @public */
export interface IExample {
  dottedImportType: import('api-extractor-lib1-test').Lib1Namespace.Inner.X;
}
```

The current implementation produces this output:

**api-extractor-scenarios/etc/test-outputs/dynamicImportType2/rollup.d.ts**
```ts
import * as Lib1Namespace_Inner_X from 'api-extractor-lib1-test';

/** @public */
export declare interface IExample {
    dottedImportType: Lib1Namespace_Inner_X.Lib1Namespace.Inner.X;
}
```

It might be more natural to name the symbol based on the module, i.e. like this:

**(proposed output)** - probably a small change to the current PR
```ts
import * as apiExtractorLib1Test from 'api-extractor-lib1-test';

/** @public */
export declare interface IExample {
    dottedImportType: apiExtractorLib1Test.Lib1Namespace.Inner.X;
}
```

Does that seem better to you?

Reply:
It sounded like you are considering to eventually further collapse it to something like below: (?)

**(possible future output?)**
```ts
import { Lib1Namespace } from 'api-extractor-lib1-test';
import X = Lib1Namespace.Inner.X;

/** @public */
export declare interface IExample {
    dottedImportType: X;
}
```

That could be an improvement, but it may be a complex feature to incorporate into API Extractor.  Thus I'd like to make sure this PR doesn't make too many assumptions about a possible future feature.

For example maybe we would instead decide to improve it to be like this:

**(another possible future output?)**
import { Lib1Namespace } from 'api-extractor-lib1-test';

/** @public */
export declare interface IExample {
    dottedImportType: Lib1Namespace.Inner.X;
}
```

Reply:
I see the point in using a camelcase `modulePath`, and I agree with it. Some paths contain non alphanumeric characters, like `from '@foo/bar/whatever.css'`, and `src/collector/Collector.ts#toAlphaNumericCamelCase` should handle them just fine.

Now `idealNameForEmit` is resolved using the following priority order:
```
singleExportName -> localName -> modulepath
```

Regarding my initial solution, I didn't put much thought in `idealNameForEmit`. I saw the pattern `singleExportName -> localName -> ?` and thought that `exportName` was the most similar token to localName. The `.` inside the name had to be removed, so I used a `.replace()` but I thought that using a clean `exportName` (after `exportPath` was extracted into it's own property) was more elegant.